dot/core/messages.go

```diff
@@ -24,19 +24,21 @@ import (
 
 // HandleTransactionMessage validates each transaction in the message and
 // adds valid transactions to the transaction queue of the BABE session
-func (s *Service) HandleTransactionMessage(msg *network.TransactionMessage) error {
+// returns boolean for transaction propagation, true - transactions should be propagoted
+func (s *Service) HandleTransactionMessage(msg *network.TransactionMessage) (bool, error) {
 	logger.Debug("received TransactionMessage")
 
 	// get transactions from message extrinsics
 	txs := msg.Extrinsics
-
+	var toPropagate []types.Extrinsic
 	for _, tx := range txs {
 		// validate each transaction
 		externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
 		val, err := s.rt.ValidateTransaction(externalExt)
 		if err != nil {
 			logger.Error("failed to validate transaction", "err", err)
-			return err
+			val.Propagate = false
```

Comment:
I think `val` will be `nil` if there's an error returned? you can remove this anyways, as it will move to the next tx

Reply:
updated.